lib/streamlit/elements/widgets/button.py

```diff
@@ -670,39 +671,45 @@ def _page_link(
         if use_container_width is not None:
             page_link_proto.use_container_width = use_container_width
 
-        # Handle external links:
-        if page.startswith("http://") or page.startswith("https://"):
-            if label is None or label == "":
-                raise StreamlitAPIException(
-                    f"The label param is required for external links used with st.page_link - please provide a label."
-                )
-            else:
-                page_link_proto.page = page
-                page_link_proto.external = True
-                return self.dg._enqueue("page_link", page_link_proto)
-
-        ctx = get_script_run_ctx()
-        ctx_main_script = ""
-        all_app_pages = {}
-        if ctx:
-            ctx_main_script = ctx.main_script_path
-            all_app_pages = ctx.pages_manager.get_pages()
-
-        main_script_directory = get_main_script_directory(ctx_main_script)
-        requested_page = os.path.realpath(
-            normalize_path_join(main_script_directory, page)
-        )
+        if isinstance(page, StreamlitPage):
+            page_link_proto.page_script_hash = page._script_hash
+            page_link_proto.page = page.title.replace(" ", "_")
+            if label is None:
+                page_link_proto.label = page.title
```

Comment:
do we have to do `replace("_", " ")` here like we do it in [the other case](https://github.com/streamlit/streamlit/pull/8692/files#diff-5fe41e0bd5c4553e4c24998cc7cd227dc719b689a9fd0a83777e29a1c2690d02R709)

Reply:
We do not because the page title is tied to the file name in the other case.